reconcile/utils/mr/base.py

```diff
@@ -206,17 +206,17 @@ def delete_tmp_branch(self, gitlab_cli):
                     f"Failed to delete branch {self.branch}. " f"Reason: {gitlab_error}"
                 )
 
-    def diffs(self, gitlab_cli):
+    def diffs(self, gitlab_cli: GitLabApi) -> Any:
         return gitlab_cli.project.repository_compare(
-            from_=self.main_branch, to=self.branch
+            from_=gitlab_cli.main_branch, to=self.branch
         )["diffs"]
 
-    def submit(self, cli: MRClient):
+    def submit(self, cli: MRClient) -> Union[Any, None]:
         if isinstance(cli, GitLabApi):
             return self.submit_to_gitlab(gitlab_cli=cli)
 
         if isinstance(cli, SQSGateway):
-            return self.submit_to_sqs(sqs_cli=cli)
+            self.submit_to_sqs(sqs_cli=cli)
 
         raise AttributeError(f"client {cli} not supported")
```

Comment:
Why did you remove `return`?

Reply:
`submit_to_sqs` does not return anything and mypy was starting to complain when i added the type hints

Reply:
But now, if someone runs it with `SQSGateway`, it will always raise an `AttributeError`.